clients/go/zbc/client.go

```diff
@@ -32,6 +35,20 @@ type ZBClientImpl struct {
 	connection     *grpc.ClientConn
 }
 
+type ZBClientConfig struct {
+	GatewayAddress      string
+	UseSecureConnection bool
+	CaCertificatePath   string
+}
+
+type ZBError string
```

Comment:
I'm not sure the `ZBError` type is all that useful if it's just an alias for string? I'm no go expert though, what's the reasoning here? How is it different from

```go
const NonExistingFileError = errors.New("expected file to exist but couldn't find anything at specified path")
```

Reply:
It's not just an alias for string because it also implements the Error method which makes it a string that also meets the error interface. Your example doesn't compile because the `errors.New(...)` isn't a constant initializer, there are no struct constants in Go. https://golang.org/ref/spec#Constants 
This goes without saying but I'm also not a Go expert, so correct me if I' wrong =P

Reply:
What you say makes sense :+1: